packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -982,7 +985,9 @@ export class GarbageCollector implements IGarbageCollector {
         }
 
         const serializedGCState = JSON.stringify(generateSortedGCState(gcState));
-        const serializedTombstones = this.tombstones.length > 0 ? JSON.stringify(this.tombstones.sort()) : undefined;
+        const serializedTombstones = this.tombstoneMode
+            ? (this.tombstones.length > 0 ? JSON.stringify(this.tombstones.sort()) : undefined)
+            : undefined;
```

Comment:
Nit: Maybe something like this can simplify this a bit, I'll leave this up to you :)

```suggestion
        const serializedTombstones = this.tombstoneMode && this.tombstones.length > 0 ? JSON.stringify(this.tombstones.sort()) : undefined;
```